app/helpers.py

```diff
@@ -0,0 +1,68 @@
+from functools import wraps
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, make_response
+
+# Decorator to check if task exists.
+def require_valid_task_id(endpoint):
+    @wraps(endpoint)
+    def fn(*args, task_id, **kwargs):
+
+        task = Task.query.get(task_id)
+
+        if not task: 
+            return make_response("", 404)
+
+        return endpoint(*args, task=task, **kwargs)
+    return fn
+
+# Decorator to check if goal exists.
+def require_valid_goal_id(endpoint):
+    @wraps(endpoint)
+    def fn(*args, goal_id, **kwargs):
+
+        goal = Goal.query.get(goal_id)
+
+        if not goal: 
+            return make_response("", 404)
+
+        return endpoint(*args, goal=goal, **kwargs)
+    return fn
+
+# Decorator to check if the request body to create a new goal includes "title".
+def require_valid_goal_request_body(endpoint):
```

Comment:
Ooh. I like this one a lot, especially the way you stacked this with the goal lookup decorator for goal PUT. Not all decorators support stacking in this way. What about how these decorators are defined lets us use them that way?